scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/util/VariableSubstitutor.java

```diff
@@ -89,11 +89,20 @@ public static String filterAndUpdate(String input,
         }
 
         String output = input;
-        for (Map.Entry<String, String> replacement : buildSubstitutes(variables).entrySet()) {
-            if (replacement.getValue() != null) {
-                output = output.replace(replacement.getKey(), replacement.getValue());
+        Map<String, String> substitutes = buildSubstitutes(variables);
+        boolean anyReplacement;
+        int depthCount = 0;
+        do {
+            anyReplacement = false;
+            depthCount++;
+            for (Map.Entry<String, String> replacement : substitutes.entrySet()) {
+                if (replacement.getValue() != null) {
+                    String newOutput = output.replace(replacement.getKey(), replacement.getValue());
+                    anyReplacement = anyReplacement || !newOutput.equals(output);
```

Comment:
could you wrap this if in a private method with a readable name to make it easy to understand?